src/plugins/intel_cpu/src/serialize.cpp

```diff
@@ -44,26 +44,39 @@ void ModelSerializer::operator<<(const std::shared_ptr<ov::Model>& model) {
     serializer.run_on_model(std::const_pointer_cast<ov::Model>(model->clone()));
 }
 
-ModelDeserializer::ModelDeserializer(std::istream & istream, model_builder fn)
-    : _istream(istream)
-    , _model_builder(fn) {
-}
+ModelDeserializer::ModelDeserializer(std::istream& istream, model_builder fn)
+    : m_istream(&istream),
+      m_model_buffer(nullptr),
+      m_model_builder(fn) {}
+
+ModelDeserializer::ModelDeserializer(const std::shared_ptr<ov::MappedMemory>& buffer, model_builder fn)
+    : m_istream(nullptr),
+      m_model_buffer(buffer),
+      m_model_builder(fn) {}
 
 void ModelDeserializer::operator>>(std::shared_ptr<ov::Model>& model) {
+    if (m_model_buffer) {
+        parse_buffer(model);
+    } else {
+        parse_stream(model);
```

Comment:
It would be better to create separate classes inherited from base `ModelDeserializer` with custom `parse()` implementation. In this case each implementations will store it's own object (`m_istream` or `m_model_buffer`)

Reply:
Done